src/components/settings/settings-custom-labels/settings-custom-labels.js

```diff
@@ -0,0 +1,282 @@
+// import createFocusDecorator from 'final-form-focus';
+import PropTypes from 'prop-types';
+import React, {
+  Component,
+  Fragment,
+} from 'react';
+import { FormattedMessage } from 'react-intl';
+import {
+  Field,
+  Form,
+} from 'react-final-form';
+import {
+  get,
+  isEqual,
+  mapKeys,
+  pickBy,
+} from 'lodash';
+
+import SafeHTMLMessage from '@folio/react-intl-safe-html';
+import {
+  Checkbox,
+  Col,
+  ConfirmationModal,
+  KeyValue,
+  RepeatableField,
+  Row,
+  TextField,
+} from '@folio/stripes/components';
+
+import SettingsForm from '../settings-form';
+
+import NavigationModal from '../../navigation-modal';
+import Toaster from '../../toaster';
+import {
+  getDisplayLabels,
+  getMatchedStringInUTF8,
+} from '../../utilities';
+
+const CUSTOM_LABELS_FIELDS = [1, 2, 3, 4, 5];
+
+export default class SettingsCustomLabels extends Component {
+  static propTypes = {
+    confirmUpdate: PropTypes.func.isRequired,
+    customLabels: PropTypes.shape({
+      errors: PropTypes.array,
+      isUpdated: PropTypes.bool,
+      items: PropTypes.object,
+    }).isRequired,
+    updateCustomLabels: PropTypes.func.isRequired,
+  };
+
+  state = {
+    modalIsOpen: false,
+    removingLabels: '',
+  }
+
+  getLabelForBool = (value) => (value
+    ? <FormattedMessage id="ui-eholdings.settings.customLabels.show" />
+    : <FormattedMessage id="ui-eholdings.settings.customLabels.hide" />);
+
+  validateLabel = (item, value, allValues) => {
+    const displayOnPublicationFinder = `${item}displayOnPublicationFinder`;
+    const displayOnFullTextFinder = `${item}displayOnFullTextFinder`;
+    const utf8Value = value ? getMatchedStringInUTF8(value) : value;
+
+    if (value && value.length > 50) {
+      return <FormattedMessage id="ui-eholdings.validate.errors.settings.customLabels.length" />;
+    } else if (utf8Value !== value) {
+      return <FormattedMessage id="ui-eholdings.validate.errors.settings.customLabels.utf" />;
+    } else if (!value && (!!allValues[displayOnFullTextFinder] || !!allValues[displayOnPublicationFinder])) {
+      return <FormattedMessage id="ui-eholdings.validate.errors.settings.customLabels.empty" />;
+    } else {
+      return undefined;
+    }
+  }
+
+  renderFields = () => (
+    <RepeatableField
```

Comment:
I don't think RepeatableField is needed since we just have 5 static fields with no add/remove logic